src/Main.elm

```diff
@@ -13,37 +14,40 @@ import Url
 
 
 main =
-    Browser.application
+    Browser.element
         { init = init
         , view = view
         , update = update
         , subscriptions = subscriptions
-        , onUrlRequest = OnUrlRequest
-        , onUrlChange = OnUrlChange
         }
 
 
 type alias Flags =
-    {}
+    { spaFlags : Ext.Browser.Navigation.SpaFlags
+    }
 
 
 type alias Model =
     { bootAt : Maybe Time.Posix
     , paragraphs : Array String
-    , navKey : Browser.Navigation.Key
+    , navKey : Ext.Browser.Navigation.Key
     , url : Url.Url
     }
 
 
 type Msg
     = TimeInput Time.Posix
     | UserInput Int String
-    | OnUrlRequest Browser.UrlRequest
+    | OnUrlRequest Ext.Browser.UrlRequest
     | OnUrlChange Url.Url
 
 
-init : Flags -> Url.Url -> Browser.Navigation.Key -> ( Model, Cmd Msg )
-init flags url navKey =
+init : Flags -> ( Model, Cmd Msg )
+init flags =
+    let
+        { url, navKey } =
+            Ext.Browser.Navigation.spaFlags flags
+    in
```

Comment:
Browser.application's init is

```elm
init : flags -> Url -> Key -> ( model, Cmd msg )
```

Browser.element's init  is

```elm
init : flags -> ( model, Cmd msg )
```

so we need to get `url` and `navKey` manually through `flags`